addons/account/models/account_move_line.py

```diff
@@ -1050,6 +1050,8 @@ def _compute_discount_allocation_needed(self):
                 },
             )
             discount_allocation_needed_vals['amount_currency'] -= discounted_amount_currency
+            if line.analytic_distribution:
+                [v.update({'analytic_distribution': line.analytic_distribution}) for v in discount_allocation_needed.values()]
```

Comment:
Please don't try to obfuscate your code
```suggestion
                for v in discount_allocation_needed.values():
                    v['analytic_distribution'] = line.analytic_distribution
```

But how does it work if there are multiple lines with the same key?
Is it just the last line that wins?

Shouldn't rather the analytic distribution be part of the key?